postprocessing/janus-pp-rec.c

```diff
@@ -300,21 +317,35 @@ int main(int argc, char *argv[])
 			return -1;
 		}
 		p->seq = ntohs(rtp->seq_number);
-		if(reset == 0) {
+		/* Due to resets, we need to mess a bit with the original timestamps */
+		if(last_ts == 0) {
 			/* Simple enough... */
 			p->ts = ntohl(rtp->timestamp);
 		} else {
-			/* Is this packet pre- or post-reset? */
-			if(ntohl(rtp->timestamp) > first_ts) {
-				/* Pre-reset... */
-				p->ts = ntohl(rtp->timestamp);
-			} else {
-				/* Post-reset... */
-				uint64_t max32 = UINT32_MAX;
-				max32++;
-				p->ts = max32+ntohl(rtp->timestamp);
+			/* Is the new timestamp smaller than the next one, and if so, is it a timestamp reset or simply out of order? */
+			if(ntohl(rtp->timestamp) < last_ts && (last_ts-ntohl(rtp->timestamp) > 2*1000*1000*1000)) {
+				reset = ntohl(rtp->timestamp);
+				JANUS_LOG(LOG_WARN, "Timestamp reset: %"SCNu32"\n", reset);
+				times_resetted++;
+				post_reset_pkts = 0;
+			} else if(ntohl(rtp->timestamp) < reset) {
+				if(post_reset_pkts < 1000) {
+					JANUS_LOG(LOG_WARN, "Updating latest timestamp reset: %"SCNu32" (was %"SCNu32")\n", ntohl(rtp->timestamp), reset);
+					reset = ntohl(rtp->timestamp);
+				} else {
+					reset = ntohl(rtp->timestamp);
+					JANUS_LOG(LOG_WARN, "Timestamp reset: %"SCNu32"\n", reset);
+					times_resetted++;
+					post_reset_pkts = 0;
+				}
 			}
+			/* Take into account the number of resets when setting the internal, 64-bit, timestamp */
+			p->ts = (times_resetted*max32)+ntohl(rtp->timestamp);
 		}
+		JANUS_LOG(LOG_INFO, "%"SCNu32" --> %"SCNu64"\n", ntohl(rtp->timestamp), p->ts);
```

Comment:
I would try to avoid this log. It dumps far to much info and is not really useful in the production. wdyt?